marathon/app.go

```diff
@@ -24,6 +29,38 @@ type App struct {
 	VersionInfo VersionInfo       `json:"versionInfo"`
 }
 
+// isExcused check if application behavior is tolerated without consequence
+func (app App) isExcused() bool {
+	if val, ok := app.Labels[applicationImmunityLabel]; ok && val == "true" {
+		return true
+	}
+	return false
+}
+
+func (app *App) substractInstance() error {
+	var instances int
+	if app.Instances >= 1 {
+		instances = app.Instances - 1
+	} else {
+		return errors.New("Unable to scale down, zero instance")
+	}
+	app.Instances = instances
+	return nil
+}
+
+func (app *App) addAppCopLabels() {
+	// rewrite labels and add new one
+	if app.Instances == 0 {
+		app.Labels["appcop"] = "suspend"
+	} else {
+		app.Labels["appcop"] = "scaleDown"
+	}
```

Comment:
This logic is copied form `substractInstance()`. How about merging this two functions into one that will take care for penalizing applications that does not behave well?